modelopt/torch/speculative/plugins/transformers.py

```diff
@@ -731,7 +746,14 @@ def _eagle_forward(
         )
         eagle_logits = eagle_lm_head(eagle_postnorm_h)
 
-        return eagle_postnorm_h, eagle_prenorm_h, eagle_logits, eagle_cache
+        draft_logits_list = [eagle_logits]
+        if self.eagle_config.parallel_draft_step > 1:
+            # Get additional draft logits from parallel draft heads
+            for draft_head in self.eagle_module.parallel_draft_heads:
```

Comment:
I think we can optimize this for-loop and do it parallelly. Can be done in following PRs perhaps.